lib/core/util.c

```diff
@@ -769,6 +769,115 @@ h2o_iovec_t h2o_build_destination(h2o_req_t *req, const char *prefix, size_t pre
     return h2o_concat_list(&req->pool, parts, num_parts);
 }
 
+#define SERVER_TIMING_DURATION_LONGEST_STR "dur=" H2O_INT32_LONGEST_STR ".000"
+
+size_t stringify_duration(char *buf, int64_t usec)
+{
+    int32_t msec = (int32_t)(usec / 1000);
+    usec -= ((int64_t)msec * 1000);
+    char *pos = buf;
+    pos += sprintf(pos, "dur=%" PRId32, msec);
+    if (usec != 0) {
```

Comment:
Can't we simply use `sprintf(pos, "%dur=%" PRId32 ".%03u", msec, (unsigned)usec)` here?

I think that always expressing in the same precision is better in terms of consistency. Or is emitting `0`s at the end prohibited by the standard?

Reply:
It's not prohibited, but there is another care for the length of header value in https://www.w3.org/TR/server-timing/:

```
To minimize the HTTP overhead the provided names and descriptions should be kept as short as possible - e.g. use abbreviations and omit optional values where possible.
```

Similarly, I thought it's better to keep the duration value as short as possible. What do you think?

Reply:
Thanks for elaborating. Let's keep the code as-is.
I can understand your point and I do not have a strong objection.